### ISSUE TITLE
UnnecessarySemicolonAfterOuterTypeDeclarationCheck reports false positive on JEP 512 compact source files

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20269]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/coding/unnecessarysemicolonafteroutertypedeclaration.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="UnnecessarySemicolonAfterOuterTypeDeclaration"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputUnnecessarySemicolonAfterOuterTypeDeclarationCompact.java <<'EOF'
void main() { }

class Inner { };
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputUnnecessarySemicolonAfterOuterTypeDeclarationCompact.java

Starting audit...
[ERROR] InputUnnecessarySemicolonAfterOuterTypeDeclarationCompact.java:3:16: Unnecessary semicolon. [UnnecessarySemicolonAfterOuterTypeDeclaration]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

UnnecessarySemicolonAfterOuterTypeDeclarationCheck must not flag the trailing `;` after `class Inner { }` in this compact file. By JEP 512 semantics `Inner` is a member of the implicit class that wraps the file, not an outer type. A trailing semicolon after a member-type declaration is in `UnnecessarySemicolonAfterTypeMemberDeclaration`'s domain, not this check's.

The check fires here because the parent of `Inner`'s `CLASS_DEF` is the root token (`COMPACT_COMPILATION_UNIT`) and the check treats every direct child of the root as an outer type, exactly as if the root were `COMPILATION_UNIT`.

Expected on the compact input: zero violations from `UnnecessarySemicolonAfterOuterTypeDeclaration`.

**Same code inside an ordinary class:**

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > InputUnnecessarySemicolonAfterOuterTypeDeclarationOrdinary.java <<'EOF'
public class InputUnnecessarySemicolonAfterOuterTypeDeclarationOrdinary {
    public void main() { }

    class Inner { };
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputUnnecessarySemicolonAfterOuterTypeDeclarationOrdinary.java

Starting audit...
Audit done.
```

**Comment by vivek-0509** [comment]
To other contributors: please do not pick this issue for now to avoid duplicate work, as I may already be working on it. There are many other open issues available, so please consider picking those and help us improve the project. Thanks for understanding!

**Comment by romani** [comment]
fix is merged
